src/pages/docs/registration-authorities/acme-for-vault.mdx

```diff
@@ -0,0 +1,266 @@
+
+[hvca (onprem)]: https://learn.hashicorp.com/tutorials/vault/pki-engine
+[hvca (cloud)]: https://cloud.hashicorp.com/products/vault
+
+## Requirements
+
+* [`step-ca`](https://smallstep.com/docs/step-ca/installation)
+* [Hashicorp Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-deploy)
+* [`jq`](https://stedolan.github.io/jq/download/) tool to parse JSON output.
+
+When not specified, we run `vault` commands as `root`. A dedicated profile will be created after configuration.
+
+## Configuration Guide
+
+### 1. Start a Vault server and log into vault
+
+For testing, you can run `vault server -dev` in a terminal window, and use an HTTP Vault address as shown here.
+Note: in dev mode, you will lose your PKI when you exit the Vault server.
+
+Now log into Vault:
+
+```bash
+export VAULT_ADDR='http://127.0.0.1:8200'
+vault login token=xxxxx
+```
+
+Output:
+
+```bash
+Success! You are now authenticated. The token information displayed below
+is already stored in the token helper. You do NOT need to run "vault login"
+again. Future Vault requests will automatically use this token.
+
+Key                  Value
+---                  -----
+token                hvs.VGBPeFkQtkGp5sDdHBug0Bge
+token_accessor       kGo8ZgkOlrw9Fa1neEjDH52m
+token_duration       ∞
+token_renewable      false
+token_policies       ["root"]
+identity_policies    []
+policies             ["root"]
+```
+
+### 2. Create your Root CA:
+
+Now run: 
+
+```bash
+ROOT_CA_COMMON_NAME="Vault Root CA"
+vault secrets enable pki
+vault secrets tune -max-lease-ttl=87600h pki
+vault write pki/root/generate/internal \
+	common_name="$ROOT_CA_COMMON_NAME" \
+	ttl=87600h
+vault read -format=json pki/cert/$(vault list -format=json pki/certs | jq -r '.[0]') | jq -r .data.certificate > root_ca.crt
+vault write pki/config/urls \
+	issuing_certificates="$VAULT_ADDR/v1/pki/ca" \
+	crl_distribution_points="$VAULT_ADDR/v1/pki/crl"
+```
+
+Your root CA is created,
+and your root CA certificate PEM is saved in `root_ca.crt`.
+
+### 3. Create your Intermediate CA:
+
+Now let's create and sign the intermediate CA using the root CA.
+
+```bash
+INTERMEDIATE_CA_COMMON_NAME="Vault Intermediate CA"
+vault secrets enable -path=pki_int1 pki
+vault secrets tune -max-lease-ttl=43800h pki_int1
+vault write -format=json pki_int1/intermediate/generate/internal \
+	common_name="$INTERMEDIATE_CA_COMMON_NAME" \
+	ttl=43800h  | jq -r .data.csr > int1.csr
+vault write -format=json pki/root/sign-intermediate \
+	csr=@int1.csr \
+	format=pem_bundle \
+	ttl=43800h | jq -r .data.certificate > int1.crt
+cat int1.crt root_ca.crt > intermediate_ca.crt
+vault write pki_int1/intermediate/set-signed \
+	certificate=@intermediate_ca.crt
+vault write pki_int1/config/urls \
+	issuing_certificates="$VAULT_ADDR/v1/pki_int1/ca" \
+	crl_distribution_points="$VAULT_ADDR/v1/pki_int1/crl"
```

Comment:
By the way, I was using the names `int1, pki_int1` because you can use a similar approach to do `CA -> INT1 -> INT2`.

Reply:
or 
```
Vault CA
     |--- INT1 -- STEP RA1
     |--- INT2 -- STEP RA2
```